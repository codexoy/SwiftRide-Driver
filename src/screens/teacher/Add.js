import React from "react"
import { View, Text, Button, StyleSheet, ScrollView } from "react-native"
import { connect } from "react-redux"
import { strings } from "../../i18n"
import Notification from "../../components/Notification"
import PageTitle from "../../components/PageTitle"
import NotificationButtons from "./NotificationButtons"

class Add extends React.Component {
	static navigationOptions = () => {
		return {
			title: "add",
			tabBarLabel: strings("tabs.add"),
			tabBarAccessibilityLabel: strings("tabs.add"),
			tabBarTestID: "AddTab"
		}
	}
	render() {
		return <ScrollView style={styles.container} />
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1
	},
	notifications: {
		flex: 1,
		paddingRight: 30,
		paddingLeft: 30,
		marginTop: 20
	},
	notification: {}
})

export default connect()(Add)
