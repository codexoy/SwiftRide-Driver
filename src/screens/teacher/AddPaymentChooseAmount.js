import React from "react"
import {
	View,
	StyleSheet,
	Text,
	TouchableHighlight,
	TouchableOpacity,
	TextInput,
	KeyboardAvoidingView,
	Platform
} from "react-native"
import { connect } from "react-redux"
import { strings } from "../../i18n"
import { Icon } from "react-native-elements"
import PageTitle from "../../components/PageTitle"
import {
	MAIN_PADDING,
	floatButtonOnlyStyle,
	DEFAULT_MESSAGE_TIME,
	fullButton
} from "../../consts"
import { API_ERROR } from "../../reducers/consts"
import { NavigationActions } from "react-navigation"
import SlidingMessage from "../../components/SlidingMessage"
import { fetchOrError } from "../../actions/utils"
import { popLatestError } from "../../actions/utils"

export class AddPaymentChooseAmount extends React.Component {
	constructor(props) {
		super(props)
		this.state = {
			amount: "000",
			slidingMessageVisible: false
		}

		this.addPayment = this.addPayment.bind(this)
	}

	componentDidUpdate() {
		const error = this.props.dispatch(popLatestError(API_ERROR))
		if (error) {
			this.setState({
				error,
				slidingMessageVisible: true
			})
		}
	}

	addPayment = async () => {
		const resp = await this.props.dispatch(
			fetchOrError("/teacher/add_payment", {
				method: "POST",
				body: JSON.stringify({
					amount: parseInt(this.state.amount),
					student_id: this.props.navigation.getParam("student")
						.student_id
				})
			})
		)
		if (resp) {
			this.setState({ error: "", slidingMessageVisible: true }, () => {
				setTimeout(
					() =>
						this.props.navigation.dispatch(
							NavigationActions.back()
						),
					DEFAULT_MESSAGE_TIME
				)
			})
		}
	}

	changeAmount = amount => {
		this.setState({ amount })
	}

	render() {
		return (
			<View style={styles.container}>
				<SlidingMessage
					visible={this.state.slidingMessageVisible}
					error={this.state.error}
					success={strings("teacher.add_payment.success")}
					close={() =>
						this.setState({ slidingMessageVisible: false })
					}
				/>
				<View style={styles.headerRow}>
					<TouchableOpacity
						onPress={() => {
							this.props.navigation.goBack()
						}}
					>
						<Icon name="arrow-forward" type="material" />
					</TouchableOpacity>
					<PageTitle
						style={styles.title}
						title={strings("teacher.add_payment.title2")}
					/>
				</View>
				<KeyboardAvoidingView
					behavior={Platform.OS === "ios" ? "position" : "height"}
					keyboardVerticalOffset={Platform.select({
						ios: 62,
						android: 24
					})}
					style={styles.container}
				>
					<View style={styles.amountContainer}>
						<TextInput
							placeholder="000.00"
							value={this.state.amount}
							onChangeText={amount => this.changeAmount(amount)}
							style={styles.amountInput}
							autoFocus={true}
							maxLength={5}
							keyboardType="number-pad"
						/>
					</View>
					<TouchableHighlight
						underlayColor="#ffffff00"
						onPress={this.addPayment}
						style={styles.floatButton}
					>
						<Text style={styles.buttonText}>{strings("done")}</Text>
					</TouchableHighlight>
				</KeyboardAvoidingView>
			</View>
		)
	}
}

function mapStateToProps(state) {
	return {
		fetchService: state.fetchService,
		errors: state.errors
	}
}
export default connect(mapStateToProps)(AddPaymentChooseAmount)

const styles = StyleSheet.create({
	container: {
		flex: 1
	},
	title: {
		marginLeft: 8,
		marginTop: -4
	},
	headerRow: {
		flexDirection: "row",
		padding: 0,
		justifyContent: "flex-start",
		marginLeft: MAIN_PADDING,
		marginRight: MAIN_PADDING,
		marginTop: 20
	},
	amountInput: {
		fontSize: 80,
		alignItems: "center",
		fontFamily: "Assistant-Light"
	},
	amountContainer: {
		borderBottomWidth: 0,
		backgroundColor: "#f4f4f4",
		padding: 20,
		marginTop: 80
	},
	buttonText: {
		color: "#fff",
		fontWeight: "bold",
		fontSize: 20
	},
	floatButton: {
		...fullButton
	}
})
