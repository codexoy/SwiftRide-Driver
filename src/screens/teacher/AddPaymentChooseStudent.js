import React from "react"
import {
	View,
	StyleSheet,
	Text,
	FlatList,
	TouchableHighlight
} from "react-native"
import { connect } from "react-redux"
import { strings } from "../../i18n"
import Row from "../../components/Row"
import UserWithPic from "../../components/UserWithPic"
import Separator from "../../components/Separator"
import { SearchBar, Button, Icon } from "react-native-elements"
import PageTitle from "../../components/PageTitle"
import { MAIN_PADDING } from "../../consts"
import { getStudents } from "../../actions/students"
import { NavigationActions } from "react-navigation"
import EmptyState from "../../components/EmptyState"
import StudentsLoader from "../../components/StudentsLoader"

export class AddPaymentChooseStudent extends React.Component {
	constructor(props) {
		super(props)
		this.state = {
			search: "",
			student: {},
			students: [],
			page: 1,
			nextUrl: "",
			loading: true
		}

		this.renderItem = this.renderItem.bind(this)
		this._getStudents()
	}

	updateSearch = search => {
		this.setState({ search, loading: true }, () => {
			this._getStudents(false)
		})
	}

	_getStudents = async (append = true) => {
		resp = await getStudents(this.props.fetchService, this.state)
		let newValue = resp.students
		if (append) {
			newValue = [...this.state.students, ...newValue]
		}
		this.setState({
			students: newValue,
			nextUrl: resp.nextUrl,
			loading: false
		})
	}

	endReached = () => {
		if (!this.state.nextUrl) return
		this.setState(
			{
				page: this.state.page + 1
			},
			() => {
				this._getStudents()
			}
		)
	}

	onPress = item => {
		this.setState(
			{
				student: item
			},
			() => {
				this.props.navigation.navigate("Second", { student: item })
			}
		)
	}

	renderItem = ({ item, index }) => {
		let noneMargin = {}
		if (index == 0) {
			noneMargin = { marginTop: 0 }
		}
		return (
			<TouchableHighlight
				underlayColor="#f9f9f9"
				key={`student${item.student_id}`}
				onPress={() => this.onPress(item)}
			>
				<Row
					leftSide={
						<View style={styles.addButton}>
							<Icon name="ios-add" type="ionicon" color="#000" />
						</View>
					}
					style={{ ...styles.row, ...noneMargin }}
				>
					<UserWithPic
						name={item.user.name}
						nameStyle={styles.nameStyle}
						width={64}
						height={64}
					/>
				</Row>
			</TouchableHighlight>
		)
	}

	_renderEmpty = () => (
		<EmptyState
			image="students"
			text={strings("no_students")}
			style={styles.empty}
		/>
	)

	_renderStudents = () => {
		if (this.state.loading) {
			return (
				<View style={styles.listLoader}>
					<StudentsLoader />
				</View>
			)
		}
		return (
			<FlatList
				data={this.state.students}
				keyboardShouldPersistTaps="always"
				renderItem={this.renderItem}
				keyExtractor={item => `student${item.student_id}`}
				onEndReached={this.endReached}
				keyboardDismissMode="on-drag"
				keyboardShouldPersistTaps="always"
				ListEmptyComponent={this._renderEmpty}
			/>
		)
	}
	render() {
		return (
			<View style={styles.container}>
				<PageTitle
					style={styles.title}
					title={strings("teacher.add_payment.title1")}
					leftSide={
						<Button
							icon={
								<Icon
									name="ios-close"
									type="ionicon"
									size={36}
								/>
							}
							onPress={() => {
								this.props.navigation.dispatch(
									NavigationActions.back()
								)
							}}
							type="clear"
							style={styles.closeButton}
						/>
					}
				/>
				<View
					style={styles.studentsSearchView}
					testID="StudentsSearchView"
				>
					<SearchBar
						placeholder={strings("teacher.students.search")}
						onChangeText={this.updateSearch}
						value={this.state.search}
						platform="ios"
						containerStyle={styles.searchBarContainer}
						inputContainerStyle={styles.inputContainerStyle}
						cancelButtonTitle={strings("teacher.students.cancel")}
						inputStyle={styles.search}
						textAlign="right"
						autoFocus={true}
						testID="searchBar"
					/>
					<Separator />
					{this._renderStudents()}
				</View>
			</View>
		)
	}
}

function mapStateToProps(state) {
	return {
		fetchService: state.fetchService
	}
}

export default connect(mapStateToProps)(AddPaymentChooseStudent)

const styles = StyleSheet.create({
	container: {
		flex: 1,
		marginLeft: MAIN_PADDING,
		marginRight: MAIN_PADDING,
		marginTop: 20
	},
	studentsSearchView: { paddingTop: 0 },
	searchBarContainer: {
		backgroundColor: "transparent",
		paddingBottom: 0,
		paddingTop: 0,
		marginTop: 16
	},
	inputContainerStyle: {
		borderRadius: 30,
		paddingLeft: 8,
		paddingTop: 6,
		paddingBottom: 6,
		width: "100%",
		marginLeft: 0,
		marginRight: 0
	},
	search: {
		alignItems: "flex-start",
		paddingLeft: 6,
		fontSize: 14,
		marginLeft: 0
	},
	nameStyle: {
		fontSize: 18,
		marginTop: 14
	},
	addButton: {
		marginTop: 12
	},
	row: { marginTop: 20 },
	closeButton: {
		marginTop: -6
	}
})
