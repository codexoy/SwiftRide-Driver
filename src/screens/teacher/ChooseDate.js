import React from "react"
import {
	View,
	Text,
	TouchableOpacity,
	StyleSheet,
	FlatList
} from "react-native"
import { connect } from "react-redux"
import { strings, dates } from "../../i18n"
import { CalendarList } from "react-native-calendars"
import ShadowRect from "../../components/ShadowRect"
import Row from "../../components/Row"
import UserWithPic from "../../components/UserWithPic"
import Separator from "../../components/Separator"
import { Icon } from "react-native-elements"
import {
	MAIN_PADDING,
	calendarTheme,
	fullButton,
	SHORT_API_DATE_FORMAT
} from "../../consts"
import Hours from "../../components/Hours"
import { getDateAndString } from "../../actions/lessons"
import EmptyState from "../../components/EmptyState"
import LessonsLoader from "../../components/LessonsLoader"
import moment from "moment"

export class ChooseDate extends React.Component {
	constructor(props) {
		super(props)
		const date = new Date()
		this.state = {
			day: date,
			selected: date.toJSON().slice(0, 10),
			items: [],
			loading: true
		}
		this._getItems(date)
		this.onDayPress = this.onDayPress.bind(this)
	}
	componentDidMount() {
		this.willFocusSubscription = this.props.navigation.addListener(
			"willFocus",
			payload => {
				this._getItems(this.state.day)
			}
		)
	}

	componentWillUnmount() {
		this.willFocusSubscription.remove()
	}

	renderArrow = direction => (
		<Icon
			name={
				direction === "left"
					? "keyboard-arrow-right"
					: "keyboard-arrow-left"
			}
			type="material"
		/>
	)
	renderItem = ({ item, index }) => {
		let student = strings("teacher.no_student_applied")
		if (item.student) {
			student = item.student.name
		}
		return (
			<Row
				key={`item${item.id}`}
				style={styles.lessonRow}
				leftSide={
					<Hours
						style={styles.hours}
						duration={item.duration}
						date={item.date}
					/>
				}
			>
				<UserWithPic
					name={student}
					user={item.student}
					nameStyle={styles.nameStyle}
					width={42}
					height={42}
				/>
			</Row>
		)
	}
	_getItems = async date => {
		const dateObject = getDateAndString(date)
		const resp = await this.props.fetchService.fetch(
			"/appointments/?is_approved=true&date=ge:" +
				dateObject.date.startOf("day").toISOString() +
				"&date=le:" +
				dateObject.date.endOf("day").toISOString(),
			{ method: "GET" }
		)
		this.setState({
			items: resp.json["data"],
			loading: false
		})
	}
	onDayPress = day => {
		this.setState(
			{
				day: day,
				selected: day.dateString,
				loading: true
			},
			() => {
				this._getItems(day)
			}
		)
	}

	_renderEmpty = () => (
		<EmptyState
			image="lessons"
			text={strings("empty_lessons")}
			imageSize="small"
		/>
	)

	_renderLessons = () => {
		if (this.state.loading) {
			return (
				<View style={styles.listLoader}>
					<LessonsLoader width={340} />
				</View>
			)
		}
		return (
			<FlatList
				ItemSeparatorComponent={() => <Separator />}
				ListEmptyComponent={this._renderEmpty}
				testID="scheduleList"
				data={this.state.items}
				renderItem={this.renderItem}
				style={styles.flatList}
				keyExtractor={item => `item${item.id}`}
			/>
		)
	}
	render() {
		return (
			<View style={styles.container}>
				<CalendarList
					current={this.state.selected || new Date()}
					style={styles.calendar}
					theme={calendarTheme}
					// Enable horizontal scrolling, default = false
					horizontal={true}
					// Enable paging on horizontal, default = false
					pagingEnabled={true}
					// Minimum date that can be selected, dates before minDate will be grayed out. Default = undefined
					minDate={new Date().getDate() - 14}
					maxDate={moment()
						.add(4, "months")
						.format(SHORT_API_DATE_FORMAT)}
					markedDates={{
						[this.state.selected]: {
							selected: true,
							disableTouchEvent: true
						}
					}}
					// Handler which gets executed on day press. Default = undefined
					onDayPress={this.onDayPress}
					// Month format in calendar title. Formatting values: http://arshaw.com/xdate/#Formatting
					monthFormat={"MMMM yy"}
					renderArrow={this.renderArrow}
					hideArrows={false}
					// Do not show days of other months in month page.
					hideExtraDays={false}
				/>
				<View style={styles.scheduleContainer}>
					<Text testID="dateString" style={styles.scheduleTitle}>
						{dates("date.formats.short", this.state.selected)}
					</Text>
					<ShadowRect style={styles.schedule}>
						{this._renderLessons()}
					</ShadowRect>
				</View>
				<TouchableOpacity
					onPress={() => {
						this.props.navigation.navigate("Lesson", {
							date: this.state.selected
						})
					}}
					style={fullButton}
					testID="continueButton"
				>
					<Text style={styles.buttonText}>
						{strings("teacher.new_lesson.continue")}
					</Text>
				</TouchableOpacity>
			</View>
		)
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1
	},
	calendar: {
		flex: 1
	},
	scheduleContainer: {
		flex: 1,
		marginTop: 20
	},
	scheduleTitle: {
		fontSize: 18,
		fontWeight: "bold",
		color: "rgb(121, 121, 121)",
		alignSelf: "flex-start",
		marginLeft: MAIN_PADDING
	},
	schedule: {
		flex: 1,
		marginTop: 24
	},
	hours: {
		marginTop: 4
	},
	nameStyle: {
		fontSize: 18,
		marginTop: 4
	},
	lessonRow: {
		marginTop: 12
	},
	buttonText: {
		color: "#fff",
		fontSize: 20,
		fontWeight: "bold"
	},
	flatList: {
		flex: 1,
		marginBottom: 40,
		alignSelf: "center",
		width: "100%"
	}
})

mapStateToProps = state => {
	return {
		fetchService: state.fetchService
	}
}
export default connect(mapStateToProps)(ChooseDate)
