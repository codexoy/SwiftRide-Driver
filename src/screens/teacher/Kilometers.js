import React, { Fragment } from "react"
import {
	View,
	StyleSheet,
	Text,
	Platform,
	TouchableOpacity,
	ScrollView,
	KeyboardAvoidingView,
	TextInput,
	Alert
} from "react-native"
import { connect } from "react-redux"
import { strings, errors } from "../../i18n"
import PageTitle from "../../components/PageTitle"
import {
	MAIN_PADDING,
	SHORT_API_DATE_FORMAT,
	fullButton,
	DATE_FORMAT
} from "../../consts"
import { Icon } from "react-native-elements"
import { fetchOrError } from "../../actions/utils"
import DateTimePicker from "react-native-modal-datetime-picker"
import moment from "moment"
import AlertError from "../../components/AlertError"
import SuccessModal from "../../components/SuccessModal"

export class Kilometers extends AlertError {
	constructor(props) {
		super(props)
		this.state = {
			date: new Date(),
			start: "",
			end: "",
			personal: "",
			successVisible: false,
			datePickerVisible: false
		}
	}

	async submit() {
		const cars = await this.props.dispatch(
			fetchOrError(`/teacher/${this.props.user.teacher_id}/cars`, {
				method: "GET"
			})
		)
		if (cars.json.data.length <= 0) {
			Alert.alert(strings("errors.title"), errors("No cars available"))
			return
		}
		const car_id = cars.json.data[0].id
		const resp = await this.props.dispatch(
			fetchOrError("/teacher/cars/" + car_id + "/kilometer", {
				method: "POST",
				body: JSON.stringify({
					date: moment(this.state.date).format(SHORT_API_DATE_FORMAT),
					start: parseInt(this.state.start),
					end: parseInt(this.state.end),
					personal: parseInt(this.state.personal)
				})
			})
		)
		if (resp) {
			this.setState({ successVisible: true })
		}
	}

	_showDateTimePicker = () => this.setState({ datePickerVisible: true })

	_hideDateTimePicker = () => this.setState({ datePickerVisible: false })

	_handleDatePicked = date => {
		this._hideDateTimePicker()
		this.setState({ date: moment(date).format(SHORT_API_DATE_FORMAT) })
	}

	changeText = (name, val) => {
		this.setState({ [name]: val })
	}

	render() {
		return (
			<View style={styles.container}>
				<SuccessModal
					visible={this.state.successVisible}
					image="lesson"
					title={strings("settings.kilometers.success_title")}
					desc={strings("settings.kilometers.success_desc", {
						date: moment(this.state.date).format(DATE_FORMAT)
					})}
					buttonPress={() => {
						this.setState({ successVisible: false })
						this.props.navigation.goBack()
					}}
					button={strings("settings.kilometers.success_button")}
				/>
				<View style={styles.headerRow}>
					<TouchableOpacity
						onPress={() => {
							this.props.navigation.goBack()
						}}
					>
						<Icon name="arrow-forward" type="material" />
					</TouchableOpacity>
					<PageTitle
						style={styles.title}
						title={strings("settings.report_types.kilometers")}
					/>
				</View>
				<ScrollView
					keyboardDismissMode={
						Platform.OS === "ios" ? "interactive" : "on-drag"
					}
					keyboardShouldPersistTaps="handled"
					style={styles.container}
				>
					<TouchableOpacity
						onPress={this._showDateTimePicker.bind(this)}
						style={styles.dateContainer}
					>
						<Text style={styles.date}>
							{moment(this.state.date).format(DATE_FORMAT)}
						</Text>
					</TouchableOpacity>
					<TextInput
						placeholder={strings(
							"settings.kilometers.start_of_day"
						)}
						value={this.state.start}
						onChangeText={val => this.changeText("start", val)}
						style={styles.input}
						keyboardType="number-pad"
					/>
					<TextInput
						placeholder={strings("settings.kilometers.end_of_day")}
						value={this.state.end}
						onChangeText={val => this.changeText("end", val)}
						style={styles.input}
						keyboardType="number-pad"
					/>
					<TextInput
						placeholder={strings("settings.kilometers.personal")}
						value={this.state.personal}
						onChangeText={val => this.changeText("personal", val)}
						style={styles.input}
						keyboardType="number-pad"
					/>
				</ScrollView>
				<KeyboardAvoidingView
					behavior={Platform.OS === "ios" ? "padding" : null}
					keyboardVerticalOffset={Platform.select({
						ios: fullButton.height,
						android: null
					})}
				>
					<TouchableOpacity
						onPress={this.submit.bind(this)}
						style={styles.floatButton}
					>
						<Text style={styles.buttonText}>
							{strings("teacher.new_lesson.done")}
						</Text>
					</TouchableOpacity>
				</KeyboardAvoidingView>
				<DateTimePicker
					isVisible={this.state.datePickerVisible}
					onConfirm={this._handleDatePicked}
					onCancel={this._hideDateTimePicker}
					maximumDate={new Date()}
					date={new Date(this.state.date)}
				/>
			</View>
		)
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1
	},
	title: {
		marginLeft: 8,
		marginTop: -4
	},
	headerRow: {
		flexDirection: "row",
		padding: 0,
		justifyContent: "flex-start",
		marginLeft: MAIN_PADDING,
		marginRight: MAIN_PADDING,
		marginTop: 20
	},
	dateContainer: {
		alignSelf: "center",
		backgroundColor: "#f4f4f4",
		borderColor: "lightgray",
		borderWidth: 1,
		padding: 12
	},
	date: { alignSelf: "flex-start", fontSize: 20, fontWeight: "bold" },
	input: {
		backgroundColor: "#f4f4f4",
		padding: 20,
		marginTop: 12,
		textAlign: "right"
	},
	buttonText: {
		color: "#fff",
		fontWeight: "bold",
		fontSize: 20
	},
	floatButton: {
		...fullButton,
		position: "relative"
	}
})
function mapStateToProps(state) {
	return {
		user: state.user,
		error: state.error,
		fetchService: state.fetchService
	}
}
export default connect(mapStateToProps)(Kilometers)
