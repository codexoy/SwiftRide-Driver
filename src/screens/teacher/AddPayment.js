import { createStackNavigator } from "react-navigation"
import AddPaymentChooseStudent from "./AddPaymentChooseStudent"
import AddPaymentChooseAmount from "./AddPaymentChooseAmount"
export default createStackNavigator(
	{ First: AddPaymentChooseStudent, Second: AddPaymentChooseAmount },
	{
		initialRouteKey: "First",
		headerMode: "none",
		navigationOptions: {
			headerVisible: false
		}
	}
)
