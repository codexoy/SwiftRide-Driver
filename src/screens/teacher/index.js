import React from "react"
import {
	createBottomTabNavigator,
	createStackNavigator
} from "react-navigation"
import Home from "./Home"
import Notifications from "./Notifications"
import ChooseDate from "./ChooseDate"
import NewLesson from "./NewLesson"
import Students from "./Students"
import NewStudent from "./NewStudent"
import Schedule from "./Schedule"
import { strings } from "../../i18n"
import { Icon } from "react-native-elements"

export default createBottomTabNavigator(
	{
		Home: Home,
		Notifications: Notifications,
		Add: {
			screen: createStackNavigator(
				{ ChooseDate: ChooseDate, NewLesson: NewLesson },
				{
					initialRouteName: "ChooseDate",
					headerMode: "none",
					navigationOptions: {
						headerVisible: false
					}
				}
			),
			navigationOptions: {
				title: "add",
				tabBarLabel: strings("tabs.add"),
				tabBarAccessibilityLabel: strings("tabs.add"),
				tabBarTestID: "NewLessonTab"
			}
		},
		Schedule: Schedule,
		Students: {
			screen: createStackNavigator(
				{ Students: Students, NewStudent: NewStudent },
				{
					mode: "modal",
					initialRouteName: "Students",
					headerMode: "none",
					navigationOptions: {
						headerVisible: false
					}
				}
			),
			navigationOptions: {
				title: "students",
				tabBarLabel: strings("tabs.students"),
				tabBarAccessibilityLabel: strings("tabs.students"),
				tabBarTestID: "StudentsTab"
			}
		}
	},
	{
		defaultNavigationOptions: ({ navigation }) => ({
			tabBarIcon: ({ focused, horizontal, tintColor }) => {
				const { routeName } = navigation.state
				let iconName
				let iconType = "material"
				if (routeName === "Home") {
					iconName = "home"
				} else if (routeName === "Notifications") {
					iconName = "notifications"
				} else if (routeName === "Add") {
					iconName = "add-circle"
				} else if (routeName === "Students") {
					iconName = "people"
				} else if (routeName == "Schedule") {
					iconName = "date-range"
				}

				// You can return any component that you like here!
				return (
					<Icon
						name={iconName}
						size={25}
						color={tintColor}
						type={iconType}
					/>
				)
			}
		}),
		tabBarOptions: {
			activeTintColor: "rgb(12,116,244)",
			inactiveTintColor: "black",
			showLabel: false
		}
	}
)
