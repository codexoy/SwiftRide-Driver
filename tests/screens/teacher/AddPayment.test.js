import "react-native"
import React from "react"

import renderer from "react-test-renderer"

import { AddPaymentChooseAmount } from "../../../src/screens/teacher/AddPaymentChooseAmount"
import { AddPaymentChooseStudent } from "../../../src/screens/teacher/AddPaymentChooseStudent"

jest.useFakeTimers()
const navigation = {
	getParam: param => {
		if (param == "student") return { student_id: 1, user: { name: "test" } }
	}
}
describe("AddPaymentChooseStudent", () => {
	test("view renders correctly", () => {
		const tree = renderer.create(<AddPaymentChooseStudent />).toJSON()
		expect(tree).toMatchSnapshot()
	})
})

describe("AddPaymentChooseAmount", () => {
	test("view renders correctly", () => {
		const tree = renderer
			.create(<AddPaymentChooseAmount navigation={navigation} />)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})
})
