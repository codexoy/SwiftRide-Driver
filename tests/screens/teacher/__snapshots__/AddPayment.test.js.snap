// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPaymentChooseAmount view renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <SuccessModal
    button="חזור לעמוד הראשי"
    buttonPress={[Function]}
    desc="תשלום של  נוסף לתלמיד test."
    image="payment"
    title="הוספת תשלום בהצלחה!"
    visible={false}
  />
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginLeft": 26,
        "marginRight": 26,
        "marginTop": 20,
        "padding": 0,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <Themed.Icon
        name="arrow-forward"
        type="material"
      />
    </TouchableOpacity>
    <PageTitle
      style={
        Object {
          "marginLeft": 8,
          "marginTop": -4,
        }
      }
      title="הוספת תשלום - סכום"
    />
  </View>
  <ScrollViewMock
    keyboardDismissMode="interactive"
    keyboardShouldPersistTaps="handled"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#f4f4f4",
          "borderBottomWidth": 0,
          "padding": 20,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoFocus={true}
        keyboardType="number-pad"
        maxLength={5}
        onChangeText={[Function]}
        placeholder="000"
        rejectResponderTermination={true}
        style={
          Object {
            "fontFamily": "Assistant-Light",
            "fontSize": 48,
            "textAlign": "right",
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <Dropdown
      absoluteRTLLayout={false}
      animationDuration={225}
      baseColor="rgba(0, 0, 0, .38)"
      containerStyle={
        Object {
          "alignSelf": "center",
          "marginTop": 20,
          "width": "90%",
        }
      }
      data={
        Array [
          Object {
            "label": "מזומן",
            "value": "cash",
          },
          Object {
            "label": "צ׳ק",
            "value": "check",
          },
          Object {
            "label": "כרטיס אשראי",
            "value": "credit",
          },
          Object {
            "label": "העברה בנקאית",
            "value": "bank",
          },
          Object {
            "label": "אחר",
            "value": "other",
          },
        ]
      }
      disabled={false}
      dropdownMargins={
        Object {
          "max": 40,
          "min": 20,
        }
      }
      dropdownOffset={
        Object {
          "left": 0,
          "top": 0,
        }
      }
      fontSize={16}
      hitSlop={
        Object {
          "bottom": 6,
          "left": 4,
          "right": 4,
          "top": 6,
        }
      }
      itemColor="rgba(0, 0, 0, .54)"
      itemCount={4}
      itemPadding={8}
      labelExtractor={[Function]}
      onChangeText={[Function]}
      propsExtractor={[Function]}
      rippleCentered={false}
      rippleDuration={400}
      rippleInsets={
        Object {
          "bottom": -8,
          "left": 0,
          "right": 0,
          "top": 16,
        }
      }
      rippleOpacity={0.54}
      rippleSequential={true}
      shadeOpacity={0.12}
      supportedOrientations={
        Array [
          "portrait",
          "portrait-upside-down",
          "landscape",
          "landscape-left",
          "landscape-right",
        ]
      }
      textColor="rgba(0, 0, 0, .87)"
      useNativeDriver={false}
      value="cash"
      valueExtractor={[Function]}
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="על מה התשלום? (ברירת מחדל: שיעורי נהיגה)"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "#f4f4f4",
          "marginTop": 12,
          "padding": 20,
          "textAlign": "right",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="ח.פ (השאר ריק אם אין)"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "#f4f4f4",
          "marginTop": 12,
          "padding": 20,
          "textAlign": "right",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      style={
        Object {
          "flexDirection": "row",
          "padding": 26,
        }
      }
    >
      <Text
        style={
          Object {
            "fontWeight": "bold",
            "marginTop": 4,
          }
        }
      >
        צור חשבונית כעת
      </Text>
      <Switch
        onValueChange={[Function]}
        style={
          Object {
            "marginLeft": "auto",
          }
        }
        value={false}
      />
    </View>
  </ScrollViewMock>
  <KeyboardAvoidingView
    behavior="padding"
    enabled={true}
    keyboardVerticalOffset={56}
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(12,116,244)",
          "borderRadius": 0,
          "bottom": 0,
          "elevation": 8,
          "height": 56,
          "justifyContent": "center",
          "position": "relative",
          "shadowColor": "rgb(12,116,244)",
          "shadowOffset": Object {
            "height": 8,
            "width": 0,
          },
          "shadowOpacity": 0.5,
          "shadowRadius": 16,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        הוסף
      </Text>
    </TouchableOpacity>
  </KeyboardAvoidingView>
</View>
`;

exports[`AddPaymentChooseStudent view renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "marginLeft": 26,
      "marginRight": 26,
      "marginTop": 20,
    }
  }
>
  <PageTitle
    leftSide={
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "marginTop": -6,
          }
        }
      >
        <Themed.Icon
          name="ios-close"
          size={36}
          type="ionicon"
        />
      </TouchableOpacity>
    }
    title="הוספת תשלום - בחירת תלמיד"
  />
  <View
    style={
      Object {
        "flex": 1,
        "paddingTop": 0,
      }
    }
    testID="StudentsSearchView"
  >
    <KeyboardAvoidingView
      behavior="padding"
      enabled={true}
      keyboardVerticalOffset={100}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <ForwardRef(Themed.SearchBar)
        autoFocus={true}
        cancelButtonTitle="ביטול"
        containerStyle={
          Object {
            "backgroundColor": "transparent",
            "marginTop": 16,
            "paddingBottom": 0,
            "paddingTop": 0,
          }
        }
        inputContainerStyle={
          Object {
            "borderRadius": 30,
            "marginLeft": 0,
            "marginRight": 0,
            "paddingBottom": 6,
            "paddingLeft": 8,
            "paddingTop": 6,
            "width": "100%",
          }
        }
        inputStyle={
          Object {
            "alignItems": "flex-start",
            "fontSize": 14,
            "marginLeft": 0,
            "paddingLeft": 6,
          }
        }
        onChangeText={[Function]}
        placeholder="חיפוש תלמיד"
        platform="ios"
        testID="searchBar"
        textAlign="right"
        value=""
      />
      <Separator />
      <View>
        <StudentsLoader />
      </View>
    </KeyboardAvoidingView>
  </View>
</View>
`;
